module/src/components/singleSelect/singleSelect.component.tsx

```diff
@@ -0,0 +1,147 @@
+import React from "react";
+import Select, { GroupBase, OnChangeValue } from "react-select";
+import SelectRef from "react-select/dist/declarations/src/Select";
+import { Form } from "../../hooks";
+import { IIcon, IconSet } from "../icon";
+import { ValidationErrors } from "../validationErrors";
+
+export type ReactSingleSelectRef = SelectRef<
+  ISelectOptionType<string>,
+  false,
+  GroupBase<ISelectOptionType<string>>
+>;
+
+export type ISelectOptionType<TSelectData = any> = {
+  value: TSelectData;
+  label: string;
+};
+
+export interface ISingleSelectProps<TSelectData = string> {
```

Comment:
We need an override option for the `value` in here, so that the component can be used without the form binder.